core/src/main/java/org/apache/iceberg/encryption/EncryptionManagerFactory.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.encryption;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.iceberg.TableMetadata;
+
+public interface EncryptionManagerFactory extends Closeable {
+
+  /**
+   * Initialize EncryptionManagerFactory from catalog properties.
+   *
+   * @param properties catalog properties
+   */
+  default void initialize(Map<String, String> properties) {}
+  /**
+   * Create encryption manager from table metadata.
+   *
+   * @param tableMetadata table metadata
+   * @return created encryption manager instance.
+   */
+  EncryptionManager create(TableMetadata tableMetadata);
```

Comment:
I think this needs to be more narrow, not `TableMetadata`.

Reply:
@rdblue , thanks a lot and agreed, at this stage we can narrow it to table properties. I am wondering if we will need table schema as well for column level encryption with different keys, cc @ggershinsky for inputs.

Reply:
@JiJiTang In general, we can set up column encryption with table properties only; but it would be good to have the schema too - this will enable us to verify the column encryption properties, making sure nothing is misspelled so we don't get an instruction to encrypt non-existing column (this can also lead to having a sensitive column remaining unencrypted..)

Reply:
@ggershinsky, the validation totally make sense, thanks a lot for the comment. I will update this method to keep only table properties and schema as params.